src/daily_output.ts

```diff
@@ -49,8 +52,15 @@ export class DailyOutput {
 	public async feedGQLData(queryResult: GQLEdgeInterface[]): Promise<void> {
 		for (const edge of queryResult) {
 			await this.aggregateData(edge);
+
+			this.currentHieght = edge.node.block.height;
+			if (edge.node.block.height && this.currentHieght !== edge.node.block.height) {
+				// only after we find a new block, because we are sure that there are no more transactions belonging to an already found bundle
+				await this.finishDataAggregation();
```

Comment:
I thin `finishDataAggregation` could change to `aggregateData` or something because it's not block specific

Reply:
I run it just after I'm done reading a block because it checks for the tips of bundles. It was previously once per run, and now once per block.

To add up: it's int a per-block basis because the data items of a bundle will all belong to the same block. Also because after each block, I now write the output with the updated rankings.

Reply:
ok, so `aggregateBlockData` is a good name

Reply:
The aggregate method would have already run at this point. This method won't do the aggregate, but to finish the computation (the diff, the weekly rewards, the ranks).

Reply:
See:
```ts
	/**
	 * From the fed data, derivate:
	 * - the ranking,
	 * - the change in percentage of the upload voume,
	 * - wether or not the group effort reached the minimum requirements,
	 * - group effort rewards, and
	 * - streak rewards
	 */
	private async finishDataAggregation(): Promise<void> {
```

Reply:
Updated: dae42464abfa17ed716ec1538318de11b14e58b2